### PR TITLE
Fixes aspnetcore#38798

### PR DESCRIPTION
**Pull request by zbecknell**
This addresses https://github.com/dotnet/aspnetcore/issues/38798.

### CODE REVIEW

File: src/BuiltInTools/dotnet-watch/HotReload/RudeEditPreference.cs [hunk 1]
```diff
@@ -19,6 +19,14 @@ public RudeEditDialog(IReporter reporter, IConsole console)
         {
             _reporter = reporter;
             _console = console;
+
+            var alwaysRestart = Environment.GetEnvironmentVariable("DOTNET_WATCH_ALWAYS_RESTART");
```
Comment (pranavkm) [inline src/BuiltInTools/dotnet-watch/HotReload/RudeEditPreference.cs:23]:
```suggestion
            var alwaysRestart = Environment.GetEnvironmentVariable("DOTNET_WATCH_RESTART_ON_RUDE_EDIT");
```

File: src/BuiltInTools/dotnet-watch/HotReload/RudeEditPreference.cs [hunk 2]
```diff
@@ -19,6 +19,14 @@ public RudeEditDialog(IReporter reporter, IConsole console)
...
+
+            if (alwaysRestart == "1" || alwaysRestart?.ToLower() == "true")
+            {
+                _reporter.Verbose($"DOTNET_WATCH_ALWAYS_RESTART = '{alwaysRestart}'. Restarting without prompt.");
```
Comment (pranavkm) [inline src/BuiltInTools/dotnet-watch/HotReload/RudeEditPreference.cs:27]:
```suggestion
                _reporter.Verbose($"DOTNET_WATCH_RESTART_ON_RUDE_EDIT= '{alwaysRestart}'. Restarting without prompt.");
```

File: src/BuiltInTools/dotnet-watch/HotReload/RudeEditPreference.cs [hunk 3]
```diff
@@ -19,6 +19,14 @@ public RudeEditDialog(IReporter reporter, IConsole console)
         {
             _reporter = reporter;
             _console = console;
+
+            var alwaysRestart = Environment.GetEnvironmentVariable("DOTNET_WATCH_ALWAYS_RESTART");
+
+            if (alwaysRestart == "1" || alwaysRestart?.ToLower() == "true")
```
Comment (pranavkm) [inline src/BuiltInTools/dotnet-watch/HotReload/RudeEditPreference.cs:25]:
```suggestion
            if (alwaysRestart == "1" || string.Equals(alwaysRestart, "true",  StringComparison.OrdinalIgnoreCase)
```

**Review by pranavkm — APPROVED** [review]

File: src/BuiltInTools/dotnet-watch/HotReload/RudeEditPreference.cs [hunk 4]
```diff
@@ -19,6 +19,14 @@ public RudeEditDialog(IReporter reporter, IConsole console)
         {
             _reporter = reporter;
             _console = console;
+
+            var alwaysRestart = Environment.GetEnvironmentVariable("DOTNET_WATCH_RESTART_ON_RUDE_EDIT");
+
+            if (alwaysRestart == "1" || string.Equals(alwaysRestart, "true",  StringComparison.OrdinalIgnoreCase)
```
Comment (zbecknell) [inline src/BuiltInTools/dotnet-watch/HotReload/RudeEditPreference.cs:25]:
```suggestion
            if (alwaysRestart == "1" || string.Equals(alwaysRestart, "true",  StringComparison.OrdinalIgnoreCase))
```

**Comment by zbecknell** [comment]
Hey @pranavkm, just added a missing closing parenthesis to your edit.

File: src/BuiltInTools/dotnet-watch/HotReload/RudeEditPreference.cs [hunk 5]
```diff
@@ -19,6 +19,14 @@ public RudeEditDialog(IReporter reporter, IConsole console)
         {
             _reporter = reporter;
             _console = console;
+
+            var alwaysRestart = Environment.GetEnvironmentVariable("DOTNET_WATCH_RESTART_ON_RUDE_EDIT");
+
+            if (alwaysRestart == "1" || string.Equals(alwaysRestart, "true",  StringComparison.OrdinalIgnoreCase))
```
Comment (pranavkm) [inline src/BuiltInTools/dotnet-watch/HotReload/RudeEditPreference.cs:25]:
```suggestion
            if (alwaysRestart == "1" || string.Equals(alwaysRestart, "true", StringComparison.OrdinalIgnoreCase))
```

**Pull request merged**